server/src/main/java/org/opensearch/cluster/coordination/Coordinator.java

```diff
@@ -1425,6 +1431,17 @@ protected void onFoundPeersUpdated() {
         }
     }
 
+    // package-visible for testing
+    synchronized void nodeCommissioned(boolean localNodeCommissioned) {
+        this.localNodeCommissioned = localNodeCommissioned;
+        peerFinder.setFindPeersInterval(localNodeCommissioned);
+    }
+
```

Comment:
So I had asked to remove the ```localNodeCommissioned``` variable as i thought we don't need it inside Coordinator for any logic. But thinking more, I think missed a case where node has stale state and if it starts the prevoting round, then it would be able to do that, so in prevoting both should be checked local state for decommissioning flag or ```localNodeCommissioned``` . @imRishN sorry for the confusion, need to add it back.

Reply:
Yes, and having it in coordinator also eases testing. Reverting the changes back